types.cc

```diff
@@ -1401,16 +1401,15 @@ tuple_type_impl::get_instance(std::vector<data_type> types) {
     return intern::get_instance(std::move(types));
 }
 
-static void validate_aux(const tuple_type_impl& t, bytes_view v, cql_serialization_format sf) {
+template <FragmentedView View>
+static void validate_aux(const tuple_type_impl& t, View v, cql_serialization_format sf) {
     auto ti = t.all_types().begin();
-    auto vi = tuple_deserializing_iterator::start(v);
-    auto end = tuple_deserializing_iterator::finish(v);
-    while (ti != t.all_types().end() && vi != end) {
-        if (*vi) {
-            (*ti)->validate(**vi, sf);
+    while (ti != t.all_types().end() && v.size_bytes()) {
+        std::optional<View> e = read_tuple_element(v);
+        if (e) {
+            (*ti)->validate(*e, sf);
         }
         ++ti;
-        ++vi;
     }
```

Comment:
Unrelated: we should check here that there is neither types nor bytes left.

Reply:
Should I do that now? Since this is a refactoring/optimization series, maybe we shouldn't try to improve the semantics of verification here.

Reply:
If you want to, but it is also fine to send as a follow-up patch.